packages/std/src/results/cosmos_msg.rs

```diff
@@ -11,6 +11,12 @@ use crate::serde::to_binary;
 
 use super::Empty;
 
+/// Like CustomQuery for better type clarity.
+/// Also makes it shorter to use as a trait bound.
+pub trait CustomMsg: Serialize + Clone + fmt::Debug + PartialEq + JsonSchema {}
```

Comment:
We added `JsonSchema` in some places in the first days of CosmWasm to make it harder for the user to put the wrong thing in there.

Reply:
That said, I would rather adapt this line in a separate PR once we agree we can use this alias in the first place

Reply:
`CustomMsg` is part of `CosmosMsg` and I think we should have consistent PartialEq and Debug for any `CosmosMsg<T>`.

Reply:
You should talk with @hashedone 

We had a number of issues about requiring trait bounds on a struct, which creeps everywhere.

CustomMsg, like CustomQuery, is implemented once per blockchain, at most. The first tests with this will show up if it doesn't implement some of these traits when they use them (in eg assert_eq!). We do require some bounds for this to compile with the entry_points, and it makes sense to do some enforcing of that.

But I think @hashedone can explain this better as he convinced me about this (that and some compiler wrangling)

Again, using CustomMsg with the same traits as currently is the scope of this PR.
Changing which traits are required by CustomMsg should be a new issue and not in this PR.